baseapp/abci.go

```diff
@@ -192,8 +192,25 @@ func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeg
 			WithHeaderHash(req.Hash)
 	}
 
+	ctx := app.deliverState.ctx
+	if app.preBeginBlocker != nil {
+		// noConsensusParams is true before migrate
+		noConsensusParams := app.GetConsensusParams(ctx) == nil
+		app.preBeginBlocker(ctx, req)
+		// Manager skips this step if Block is non-nil since upgrade module is expected to set this params
+		// and consensus parameters should not be overwritten.
+		if noConsensusParams {
```

Comment:
The modified check here is change from nil to not nil, do you mean change from value A to value B?

Reply:
"if consensus params is modified in preBeginBlocker, reload it", do you agree with this logic? the purpose is to make sure the following logic always sees the latest consensus parameters, no matter if it's loaded before, right? 
if it's loaded before, we need to refresh it, because it's not the latest, if it's modified in `preBeginBlocker`, right?

Reply:
since we name it as `preBeginBlocker`, it's more generic than migration. `preBeginBlocker` is something that:
- runs before begin blocker
- allowed to modify consensus parameters, and the changes are visible to the following state machine logics.